desc/objectives/_stability.py

```diff
@@ -367,3 +369,246 @@ def compute(self, params, constants=None):
+        Not used since the growth rate is always normalized.
+    normalize_target : bool, optional
+        Whether target and bounds should be normalized before comparing to computed
+        values. If `normalize` is `True` and the target is in physical units,
+        this should also be set to True.
+    grid : Grid, optional
+        Collocation grid containing the nodes to evaluate at.
+    name : str, optional
+        Name of the objective function.
+
+    """
+
+    _coordinates = "r"
+    _units = "(normalized)"
+    _print_value_fmt = "Ideal-ballooning Stability: {:10.3e} "
+
+    def __init__(
+        self,
+        eq=None,
+        target=None,
+        bounds=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        deriv_mode="rev",
+        loss_function=None,
+        rho=0.5,
+        alpha=jnp.linspace(0, jnp.pi, 8)[:, None],
+        zetamax=3 * jnp.pi,
+        nzeta=200,
+        name="ideal-ball gamma",
+    ):
+        if target is None and bounds is None:
+            target = 0
+
+        self.rho = rho
+        self.alpha = alpha
+        self.zetamax = zetamax
+        self.nzeta = nzeta
+
+        super().__init__(
+            things=eq,
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            loss_function=loss_function,
+            deriv_mode=deriv_mode,
+            name=name,
+        )
+
+    def build(self, eq=None, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        eq : Equilibrium, optional
+            Equilibrium that will be optimized to satisfy the Objective.
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        eq = self.things[0]
+
+        # we need a uniform grid to get correct surface averages for iota
+        iota_grid = LinearGrid(rho=self.rho, M=eq.M_grid, N=eq.N_grid, NFP=eq.NFP)
+        self._iota_keys = ["iota", "iota_r", "shear"]  # might not need all of these
+        iota_profiles = get_profiles(self._iota_keys, obj=eq, grid=iota_grid)
+        iota_transforms = get_transforms(self._iota_keys, obj=eq, grid=iota_grid)
+
+        ## Separate grid to calculate the right length scale for normalization
+        len_grid = LinearGrid(rho=1.0, M=eq.M_grid, N=eq.N_grid, NFP=eq.NFP)
+        self._len_keys = ["a"]
+        len_profiles = get_profiles(self._len_keys, obj=eq, grid=len_grid)
+        len_transforms = get_transforms(self._len_keys, obj=eq, grid=len_grid)
+
+        # make a set of nodes along a single fieldline
+        zeta = np.linspace(-self.zetamax, self.zetamax, self.nzeta)
+        rho, alpha, zeta = np.reshape(
+            np.broadcast_arrays(self.rho, self.alpha, zeta), (3, -1)
```

Comment:
I think you might actually want `np.meshgrid` here rather than `np.broadcast_arrays`? then you don't need to give the extra dimension to alpha as mentioned above.

Reply:
The `Grid.create_meshgrid().nodes` may be useful

Reply:
What I have does the job. Do you have a strong argument against broadcast arrays?

Reply:
mainly that it requires alpha to have that extra dimension which feels clunky (and if you make the extra dimension at the beginning by accident then it messes up the order of the nodes which will break other stuff), and would only work for a single rho value when it seems like the rest of the code should in theory work for multiple rho (assuming we get the vectorized thing working). I can try fixing along with vectorization

Reply:
Refactored everything.